packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -5,7 +5,7 @@
 
 import {Getter} from '@loopback/context';
 import assert from 'assert';
-import legacy from 'loopback-datasource-juggler';
+import legacy, {ObserverMixin} from 'loopback-datasource-juggler';
```

Comment:
I think we are intentionally keeping juggler types very separated from LB4 types. Take a look at lines 41-53, where we are re-exporting a subset of legacy types in `juggler` namespace.

Please make the following changes:

- Revert this line back.
- Re-export `legacy.ObserverMixin` in `juggler` namespace 
- Modify all places below using `ObserverMixin` to use `juggler.Mixin`

Please add a comment to all those modified places to mention that it's a temporary workaround and add a link to the GitHub issue tracking the follow-up task of improving juggler typings to correctly include ObserverMixin in `PersistedModelClass` API.

Reply:
👍